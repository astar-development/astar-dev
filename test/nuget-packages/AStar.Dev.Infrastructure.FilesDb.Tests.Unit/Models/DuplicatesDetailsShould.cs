using AStar.Dev.Utilities;
using JetBrains.Annotations;

namespace AStar.Dev.Infrastructure.FilesDb.Models;

[TestSubject(typeof(DuplicatesDetails))]
public class DuplicatesDetailsShould
{
    [Fact]
    public void ContainTheExpectedProperties()
        => new DuplicatesDetails
           {
               Id                 = 1,
               ImageDetailsId     = 1,
               DirectoryName      = "MockDirectoryName",
               SoftDeleted        = true,
               FileName           = "MockFileName",
               FileSize           = 1234,
               FileHandle         = "MockFileHandle",
               LastViewed         = new DateTimeOffset(new (2025, 6, 28, 22, 20, 37, DateTimeKind.Utc)),
               IsImage            = true,
               DetailsLastUpdated = new (new (2025, 6, 28, 22, 30, 37, DateTimeKind.Utc)),
               HardDeletePending  = true,
               MoveRequired       = true,
               SoftDeletePending  = true,
               Height             = 456,
               Width              = 123,
               Instances          = 987
           }
           .ToJson().ShouldMatchApproved();
}
