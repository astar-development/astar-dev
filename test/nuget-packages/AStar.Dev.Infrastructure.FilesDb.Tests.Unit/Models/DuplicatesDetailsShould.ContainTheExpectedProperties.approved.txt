{
  "id": 1,
  "imageDetailsId": 1,
  "fileName": "MockFileName",
  "directoryName": "MockDirectoryName",
  "height": 456,
  "width": 123,
  "fileSize": 1234,
  "fileHandle": "MockFileHandle",
  "isImage": true,
  "instances": 987,
  "detailsLastUpdated": "2025-06-28T22:30:37+00:00",
  "lastViewed": "2025-06-28T22:20:37+00:00",
  "softDeleted": true,
  "softDeletePending": true,
  "moveRequired": true,
  "hardDeletePending": true
}